File: leetcode/code/461.hamming_distance.go
package main

import "fmt"

func hammingDistance(x int, y int) int {
    xor := x ^ y
    result := 0

    for xor != 0 {
        xor &= xor - 1
        result++
    }

    return result
}


func main() {
    if hammingDistance(1, 4) != 2 {
        fmt.Println("code here should never be executed")
    }
}

File: leetcode/code/461.hamming_distance.py
class Solution(object):
    def hammingDistance(self, x, y):
        xor = x ^ y
        result = 0

        while xor != 0:
            xor &= xor - 1
            result += 1

        return result


if __name__ == "__main__":
    assert Solution().hammingDistance(1, 4) == 2, "your buggy code"
